src/sage/groups/libgap_wrapper.pyx

```diff
@@ -362,7 +362,7 @@ class ParentLibGAP(SageObject):
                 for gap_subgroup in self._libgap.MaximalNormalSubgroups()]
 
     @cached_method
-    def gens(self):
+    def gens(self) -> tuple:
```

Comment:
This comment applies to all your uses of the tuple/list annotation in this PR: for any of these, are you able to specify what type the tuple/list is composed of? Even if you can't get more precise than a fairly general class like `Element`, it would still be better than just a generic tuple. You can specify a tuple of for example three `Element`s by `tuple[Element, Element, Element]` or an unknown number of `Element`s by `tuple[Element, ...]`. Of course the most precise object you can use would be ideal. If there are really absolutely no guarantees on the type then I guess that's fine (but surely at least `SageObject` would work?).